allen-cell-animated/nucmorph-colorizer
# feat: Show recent datasets in load dropdown

Problem
=======
Closes #122!

Solution
========
- Clicking on the input field in the Load menu will show a list of the 10 most recently opened datasets, with the most recent datasets at the top.
- Successfully opened datasets are saved to local storage (using the new `usehooks-ts` library), so they persist between browser sessions.

Misc:
- Adds `assets/index.ts` to start isolating some weird import syntax to a single location, specifically for SVG icon used in `LabeledDropdown`.
- Caught some typos in documentation elsewhere in the project.
- Fixes a bug where `fetchWithTimeout` would not actually time out (was able to catch this due to today's Vast outage 😋)
  - Made a new issue for adding unit tests for this (#142)


## Type of change
* New feature (non-breaking change which adds functionality)


Change summary:
---------------
* Shows a list of the 10 most recently opened datasets in the Load modal, adding open/close behavior.
* Tweaks/improvements to Load modal UI.
* Adds the `usehooks-ts` library.
* Fixes issue with `fetchWithTimeout` where it would not time out.

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-141/
1. Go to the Load menu, and load in a dataset (https://dev-aics-dtp-001.int.allencell.org/dan-data/colorizer/data/test)
1. Open the load menu again. There should now be two options in the recent datasets dropdown.
1. Clicking an option in the dropdown will populate it in the input.
1. Open another dataset (https://dev-aics-dtp-001.int.allencell.org/dan-data/colorizer/EMT/collection.json).
1. Test switching between datasets and loading each one. The list should be ordered by most recent still.

Screenshots (optional):
-----------------------
![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/85326520-028a-4284-bfa6-e7efcf1a4888)

![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/1cb220f2-9f80-48c9-a3b8-cda00d3b2aec)

🔉 (Turn on audio for walkthrough!)

https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/5cfcd484-e63f-4e26-8b83-a28baaf2fa6c


Keyfiles (delete if not relevant):
-----------------------
1. `LoadDatasetButton.tsx`

Thanks for contributing!
